### DISCUSSION TITLE
Support Linux board

### DISCUSSION THREAD

**Discussion by SecT0uch** [kizniche/Mycodo#1470] [General]
Would you consider extending the list of compatible board ?

I'm a [MudPi](https://github.com/mudpi/mudpi-core) user and contributor. And I've helped to [extend support for other Linux SBC](https://github.com/mudpi/mudpi-core/issues/15), you can see the PR [here](https://github.com/mudpi/mudpi-core/pull/16).

The main task has been to get rid of Pi-only modules (RPi.GPIO) and migrate to [Adafruit Blinka](https://github.com/adafruit/Adafruit_Blinka).

It now runs  on Raspberry Pi, Pine64 and technically all the blinka-compatible boards.

For example I'm running it on a Sopine baseboard on armbian. I got a couple of sensors and a relay module, and it works flawlessly.

**Comment by kizniche** [comment]
I'm not entirely comfortable with expanding support for different hardware. One of the best parts of about developing software on one hardware platform is compatibility. Pis are also so inexpensive and abundant. Do you have any specific desire to run Mycodo on another SBC that provides something a Pi cannot?

**Comment by kizniche** [comment]
This issue has been mentioned on **DIY Automation Forum**. There might be relevant details there:

https://forum.kylegabriel.com/t/mycodo-on-other-non-pi-single-board-computers/119/1

**Comment by SecT0uch** [comment]
> I'm not entirely comfortable with expanding support for different hardware. One of the best parts of about developing software on one hardware platform is compatibility. Pis are also so inexpensive and abundant. Do you have any specific desire to run Mycodo on another SBC that provides something a Pi cannot?

I think being able to run Mycodo on [70+ boards models](https://circuitpython.org/blinka) would be a big plus, to have a larger community, and potentially, more contributors. 
Each board has it's own specificities and it would add some flexibility. (ie: some boards have better performances, you could run on eMMC, etc.)

**Comment by jmonday** [comment]
I've been wanting to experiment with Mycodo for months now. I haven't been able to because I don't have a Raspberry Pi and it's very difficult finding them right now. I understand wanting to support a single hardware platform but it does make it difficult to get started when said hardware isn't easily available.
